external/glfw/js/library_glfw.js

```diff
@@ -563,7 +563,7 @@ var LibraryGLFW = {
 
     refreshJoysticks: function(forceUpdate) {
         // Produce a new Gamepad API sample if we are ticking a new game frame, or if not using emscripten_set_main_loop() at all to drive animation.
-        if (forceUpdate || Browser.mainLoop.currentFrameNumber !== GLFW.lastGamepadStateFrame || !Browser.mainLoop.currentFrameNumber) {
+        if (GLFW.isGamepadCallbackSet &&(forceUpdate || Browser.mainLoop.currentFrameNumber !== GLFW.lastGamepadStateFrame || !Browser.mainLoop.currentFrameNumber)) {
```

Comment:
Oof, feels a bit complex. Can't we split this into an early exit check first and then keep the rest as it was?

```
if (!GLFW.isGamepadCallbackSet) {
   return;
}
if (forceUpdate || Browser.mainLoop.currentFrameNumber !== GLFW.lastGamepadStateFrame || !Browser.mainLoop.currentFrameNumber) {
   ... do stuff
```